scripts/detect_missing_i18n.py

```diff
@@ -0,0 +1,224 @@
+        f':{longest_line_count}:{longest_line}'
+    )
+
+    errcount: int = 0
+    warnings: int = 0
+
+    for file in valid_files:
+
+        contents = file.read_text()
+        lines = contents.splitlines()
+
+        for line_index, line in enumerate(lines):
+            line_number = line_index + 1
+
+            includes_error_element = re.search(i18n_element_missing_regex, line)
+            includes_warn_element = re.search(i18n_element_warn_regex, line)
+            includes_error_attribute = re.search(i18n_attr_missing_regex, line)
+            includes_warn_attribute = re.search(i18n_attr_warn_regex, line)
+
+            # Check first if the line contains any instances at all of the offending regexes.
+            if not (
+                includes_error_element
+                or includes_warn_element
+                or includes_error_attribute
+                or includes_warn_attribute
+            ):
+                continue
+
+            # Find the exact index within the line where the regex is matched.
+            for char_index in range(len(line)):
+                position = char_index + 1
+                search_chunk = line[char_index:]
+
+                if "<!--" in line[:char_index]:
+                    break
+
+                untranslated = False
+
+                # Element with untranslated elements that isn't encased in a translation function
+                if re.match(i18n_element_missing_regex, search_chunk) and not (
+                    re.match(i18n_substring_regex, search_chunk)
+                    and "$:" in line[:char_index]
+                ):
+                    errtype = err_prefix
+                    errcount += 1
+                    untranslated = True
+
+                # Element with bypassed elements that isn't encased in a translation function
+                elif re.match(i18n_element_warn_regex, search_chunk) and not (
+                    re.match(i18n_substring_regex, search_chunk)
+                    and "$:" in line[:char_index]
+                ):
+                    errtype = warn_prefix
+                    warnings += 1
+                    untranslated = True
+
+                # Element with untranslated attributes
+                elif re.match(i18n_attr_missing_regex, search_chunk):
+                    errtype = err_prefix
+                    errcount += 1
+                    untranslated = True
+
+                # Element with bypassed attributes
+                elif re.match(i18n_attr_warn_regex, search_chunk):
+                    errtype = warn_prefix
+                    warnings += 1
+                    untranslated = True
+
+                if untranslated:
+                    filestring = f'{file}:{line_number}:{position}'
+                    padding = spacing_base - len(filestring) + 4
+                    print(
+                        f"{errtype} \033[4m{filestring}\033[0m{' ' * padding}{search_chunk}"
+                    )
```

Comment:
Oh, a few tweaks on this one!

- We generally use `\t` between columns
- Cool TIL there's an underline terminal code 😁 Can we put that in a helper function to make this a little easier to read?
- We can use ljust to avoid a bit of the math there :+1: 

print('\t'.join([errtype, terminal_underline(filestring).ljust(spacing_base + 12, search_chunk])